packages/drivers/routerlicious-driver/src/test/r11sSocketTests.spec.ts

```diff
@@ -69,129 +74,191 @@ describe("R11s Socket Tests", () => {
+			});
+			await assert.rejects(
+				mockSocket(socket as unknown as Socket, async () =>
+					documentService.connectToDeltaStream(client),
+				),
+				{
+					errorType: expectedErrorType,
+					scenarioName: errorEventName,
+					internalErrorCode: expectedInternalErrorCode,
+				},
+				"Error should have occurred",
+			);
 		});
+	}
 
-		await assert.rejects(
-			mockSocket(socket as unknown as Socket, async () =>
+	function runSocketErrorTest(
+		description: string,
+		errorToThrow: any,
+		expectedErrorType: string,
+		expectedInternalErrorCode: string,
+	) {
+		it(description, async () => {
+			const errorEventName = "connect_document_success";
+			socket = new ClientSocketMock({
+				connect_document: { eventToEmit: errorEventName },
+			});
+			const connection = await mockSocket(socket as unknown as Socket, async () =>
 				documentService.connectToDeltaStream(client),
-			),
-			{
-				errorType: DriverErrorTypes.authorizationError,
-				scenarioName: "connect_document_error",
-				internalErrorCode: "TokenRevoked",
-			},
-			"Error should have occurred",
-		);
-	});
-
-	it("Socket error with Token Revoked error", async () => {
-		const errorToThrow = {
-			code: 403,
-			message: "TokenRevokedError",
-			retryAfterMs: 10,
-			internalErrorCode: "TokenRevoked",
-			errorType: DriverErrorTypes.authorizationError,
-			canRetry: false,
-		};
-		const errorEventName = "connect_document_success";
-		socket = new ClientSocketMock({
-			connect_document: { eventToEmit: errorEventName },
+			);
+			let error: IAnyDriverError | undefined;
+			connection.on("disconnect", (reason: IAnyDriverError) => {
+				error = reason;
+			});
+			socket.sendErrorEvent(errorToThrow);
+			assert(
+				error?.errorType === expectedErrorType,
+				`Error type should be ${expectedErrorType}`,
+			);
+			assert(error.scenarioName === "error", "Error scenario name should be error");
+			assert(
+				(error as any).internalErrorCode === expectedInternalErrorCode,
+				`Error internal code should be ${expectedInternalErrorCode}`,
+			);
 		});
+	}
 
-		const connection = await mockSocket(socket as unknown as Socket, async () =>
-			documentService.connectToDeltaStream(client),
-		);
-		let error: IAnyDriverError | undefined;
-		connection.on("disconnect", (reason: IAnyDriverError) => {
-			error = reason;
+	function runClientErrorTest(
+		description: string,
+		clientError: any,
+		expectedErrorType: string,
+	) {
+		it(description, async () => {
+			const socketEventName = "connect_document_success";
+			socket = new ClientSocketMock({
+				connect_document: { eventToEmit: socketEventName },
+			});
+			const connection = await mockSocket(socket as unknown as Socket, async () =>
+				documentService.connectToDeltaStream(client),
+			);
+			const disconnectEventP = new Promise<{ clientId: string; errorType: string }>(
+				(resolve) => {
+					assert(socket !== undefined, "Socket should be defined");
```

Comment:
Unnecessary in test code? We know we're always defining the socket a few lines above.

Reply:
When I remove this assert, the socket object below is it erroring saying its potentially undefined

Reply:
Ah, it's a suite-scoped variable, not a local one... yeah, this is fine then.